handlers/aws/utils.py

```diff
@@ -510,3 +529,39 @@ def expand_event_list_from_field_resolver(integration_scope: str, field_to_expan
 
 def _get_hex_prefix(src: str) -> str:
     return hashlib.sha3_384(src.encode("UTF8")).hexdigest()
+
+
+@dataclass
+class ARN(object):
+    """The Amazon Resource Name (ARN) of an AWS resource."""
+
+    id: str
+    service: str
+    region: str
+    account_id: str
+
+
+def anonymize_arn(arn: str) -> ARN:
+    """Anonymize an ARN by hashing the account ID and the resource ID."""
+    arn_components = arn.split(":")
+
+    hashed_arn = hashlib.sha256(arn.encode("utf-8")).hexdigest()[:10]
```

Comment:
hello @zmoog , beware that we experienced collision when truncating the sha256 hash to 10 chars, better use the full value every where we apply the hash